pkg/log/zap/zap_test.go

```diff
@@ -459,4 +459,81 @@ var _ = Describe("Zap log level flag options setup", func() {
 
 	})
 
+	Context("with only -zap-devel flag provided", func() {
+		It("Should set dev=true.", func() {
+			args := []string{"--zap-devel=true"}
+			fromFlags.BindFlags(&fs)
+			if err := fs.Parse(args); err != nil {
+				Expect(err).ToNot(HaveOccurred())
+			}
+			out := Options{}
+			UseFlagOptions(&fromFlags)(&out)
+
+			Expect(out.Development).To(BeTrue())
+			Expect(out.Encoder).To(BeNil())
+			Expect(out.Level).To(BeNil())
+			Expect(out.StacktraceLevel).To(BeNil())
+			Expect(out.EncoderConfigOptions).To(BeNil())
+		})
+		It("Should set dev=false with log options.", func() {
+			args := []string{"--zap-log-level=2"}
```

Comment:
This seems to be setting `args` without `-zap-devel`, but the context it is in says `with only -zap-devel flag provided`. That doesn't seem quite right.